LSPS5/README.md

```diff
@@ -0,0 +1,512 @@
+webhook interface.
+
+Parameters are required unless otherwise stated.
+
+* `lsps5.webhook_registered` - The client has just
+  recently successfully called the `lsps5.set_webhook`
+  API.
+  No parameters `{}`.
+* `lsps5.payment_incoming` - The client has one or
+  more payments pending to be received.
+  No parameters `{}`.
+* `lsps5.expiry_soon` - There is an HTLC or
+  other time-bound contract, in either direction, on
+  one of the channels between the client and the LSP,
+  and it is within 24 blocks of being timed out, and
+  the timeout would cause a channel closure.
+  Parameters:
+  * `timeout` - The blockheight at which the LSP would
+    be forced to close the channel in order to enforce
+    the HTLC or other time-bound contract.
+* `lsps5.liquidity_management_request` - The LSP wants
+  to take back some of the liquidity it has towards the
+  client, for example by closing one or more of the
+  channels it has with the client, or by splicing out.
+  No parameters `{}`.
+* `lsps5.fees_change_incoming` - The LSP wants to change
+  feerates, either for the LSP-to-client channel(s),
+  or for other auto-management services.
+  Parameters:
+  * `direction` - A rough estimate of the direction
+    of the fees, one of the following strings:
+    `"lower"`, `"mixed"`, `"higher"`.
+* `lsps5.goodbye` - The LSP wants to end all services to
+  the client, and close all channels with the client.
+  The LSP is likely to no longer accept channel opening
+  and other requests from the client.
+  No parameters `{}`.
+* `lsps5.onion_message_incoming` - The client has one
+  or more [onion message][]s pending to be received.
+  No paramters `{}`.
+
+[onion message]: https://github.com/lightning/bolts/blob/bccab9afc269a50a579850e71b06e58ef16c4e92/04-onion-routing.md#onion-messages
+
+Future revisions of this LSPS MAY define new notification
+methods.
+Notification delivery services MUST ignore any notification
+method it does not recognize, and SHOULD log unrecognized
+notification methods as an unusual event.
+
+Future revisions of this LSPS MAY define new parameters for
+existing notification methods.
+Notification delivery services MUST ignore any parameters it does
+not recognize, and SHOULD log unrecognized parameters as an
+unusual event.
+
+### Implementation Quality Guidelines
+
+> **Non-normative** While these guidelines are not technically
+> required in order to comply with this standard,
+> production-quality, non-beta software should really follow
+> these guidelines.
+
+#### Mobile Client: Privacy On Mobile OS Notification
+
+[There are claims that governments have used push notifications
+to spy on users][push-notif-surveill].
+
+[push-notif-surveill]: https://web.archive.org/web/20231211010610/https://www.reuters.com/technology/cybersecurity/governments-spying-apple-google-users-through-push-notifications-us-senator-2023-12-06/
+
+To avoid this, a quality implementation of an LSPS5 notification
```

Comment:
My understanding is that this will be left up to the client, as the client vendor would create the notification delivery server, which encrypts the LSP notification before giving it to the client.  Do we want end-to-end encryption, including encrypting the LSP node ID from the notification delivery server provided by the client vendor?

Reply:
Basically:

* A single developer provides two pieces:
 * Code for a client
 * A notification delivery service

Any encryption would protect only high-entropy information. So e.g. stuff like "you have an incoming payment" is common and surveillors can assume that 90% of notifications are that form.

What needs to be protected, if the client is not willing to report it to surveillors, is the identity of the LSP. Assuming there is a healthy ecosystem of multiple competing LSPs, then that is more "high entropy" than "you have an incoming payment".

However, the client can always leak the identity of the LSP to surveillors, even if there is LSP->notification delivery service -> mobile OS server -> client encryption.  Which is why I decided to leave out a cryptographic requirement here.

If ever, what do you propose specifically? If the notification delivery service knows the LSP identity anyway (which it might need to know in order to prevent non-LSP spammers from spamming the client and draining its battery) then you still have the issue that the notification delivery service leaks it in the clear to the mobile OS server (at which point you are already surveilled).  If we want to remove the LSP identity from the notification delivery service, then the client needs to issue some kind of bearer token that the LSP presents to the notification delivery service, what exactly would you propose?

Reply:
In February 1 meeting, it was agreed to leave out any encryption specification.